### ISSUE TITLE
Bug: disabled movement persisting between saves

### ISSUE THREAD

**Issue by prurigro** [00-Evan/shattered-pixel-dungeon#14]
Hey there!

I've been playing the git-master builds of shattered-pd since you started pushing commits for 0.2.3f, and overall it's been great! You've fixed the few issues I'd noticed in previous releases, and now that the second Quick Slot seems to be in working order, I've been able to test some of the dual-item-combo characters I'd previously thought up (but found too annoying with a single Quick Slot to fully embrace).

That said, I've run into a bug that I initially hit on one of the earlier 0.2.3f commits, and it's still present after updating to your latest push. It was triggered at some point while rapidly flipping between the inventory and attempting to move with an enemy on screen, so I'm not sure if this is new or simply difficult to produce. Once triggered, my character became unable to move (tapping a location on the screen does nothing), while zooming, scrolling and item use continue to work as expected. I hit a similar bug with pixel-dungeon-gdx a while back that I traced to a divide by zero error, but I feel like this is more cooked into the logic somehow as I can quit the game, kill shattered-pixel-dungeon's process, start it back up, load my save and the inability to move remains.

Here's a copy of my shattered-pd data folder in the hope that you might be able to see what's up: http://96.126.108.7:90/shatteredpd-nomovesmage.tar.gz

Cheers!

**Comment by prurigro** [comment]
Hmm, it looks like you merged a whole whack of changes a minute ago! Let me try a build with those to make sure the issue hasn't been resolved.

**Comment by 00-Evan** [comment]
Hey, um.. I just checked your save file, your character is rooted.

Wait a couple turns and try moving again.

**Comment by prurigro** [comment]
Oh wow-- you're right! The game had lagged up and suddenly I was unable to move in a slightly different location, and since I hadn't seen my character interact with an enemy before getting stuck, I didn't even think to consider that this might be the cause... I think my phone must have queued up a bunch of operations and then played them out faster than I could see, but I still jumped the gun by not doing some basic "is it plugged in" tests before filing this report. Sorry about that! (Time to switch back to the default I/O schedule on that note)

On the bright side, this means I've done a few play throughs issue-free :) The only thing I noticed, and I only mention it in case this wasn't your intention, but I've had one thief drop an armband each play through so far, which might mean the odds of finding one are good, or maybe I just lucked out.
